### ISSUE TITLE
Improve submission process with one more step for Context selection

### ISSUE THREAD

**Issue by fpietrosanti** [globaleaks/GlobaLeaks#354]
his ticket is to implement a STEP-1 in the submission interface that does provide the context selection, including the description.

**Comment by evilaliv3** [comment]
@fpietrosanti with commit https://github.com/globaleaks/GlobaLeaks/commit/a9b246cbb701eb3d748643772128ec29408b0987 i've implemented what you see in the screenshots.

for me it's the short term fix in order to make the interface more intuitive but now i've implemented it i'm not sure it was what you had in mind.

if your was a different idea that would require impacting on current deploymend (think e.g. to publeaks/alac as reference for compatibility with your idea) let's put this ticket in whishlist as we would not be able to address more other changes before taliking a bit with the adopters.

![screenshot from 2015-05-27 17 03 44](https://cloud.githubusercontent.com/assets/217034/7840572/4e61357e-0498-11e5-855a-17b134b64b61.png)

**Comment by fpietrosanti** [comment]
@evilaliv3 maybe the default layout should provide a different "visualization" that's more user friendly. For example the user must be shown all the context without having to click to list them

**Comment by evilaliv3** [comment]
sure. this would involve more invasive change of the minimum fix i've currently applied.
let's postpone the discussion for the moment.

**Comment by fpietrosanti** [comment]
related to #405

**Comment by evilaliv3** [comment]
I've quite implemented this, it wount appear as a "step 1" in the tab selection as all the other tab depends on the context selections (i.e. the presence or not of the recipients step and the number and the question of the steps is know only after context selection)

**Comment by evilaliv3** [comment]
Small cards:
![screenshot from 2016-03-19 19 21 55](https://cloud.githubusercontent.com/assets/217034/13900506/5c9a3a0a-ee08-11e5-9541-1921fdce5b58.png)

Big cards:
![screenshot from 2016-03-19 19 23 54](https://cloud.githubusercontent.com/assets/217034/13900509/695d4232-ee08-11e5-83bb-799f1c97a6ef.png)

**Comment by fpietrosanti** [comment]
If we think the use-case to be like "list of dozens or hundreds of public agencies", probably having something more like a "list" with a title and small description listed vertically may be better, but that's probably more of an UX optimization 
That's an example on how alertline have a workflow for multiple "context" selection https://essabank.alertline.com/gcs/welcome (but this may goes with/require #405), so it maybe a future extension (when we'll additionally include default-provided questionnaire database)
<img width="971" alt="screen shot 2016-03-19 at 7 41 48 pm" src="https://cloud.githubusercontent.com/assets/770363/13900627/eafd56d6-ee0a-11e5-9881-48aae01ae13c.png">

**Comment by evilaliv3** [comment]
![screenshot from 2016-03-20 19 08 51](https://cloud.githubusercontent.com/assets/217034/13906125/54ce3b42-eecf-11e5-930e-0248edbcb883.png)

here we go!

**Comment by evilaliv3** [comment]
implemented!
